File: easy/3908_valid_digit_number/solution.cpp
class Solution {
public:
  bool validDigit(int n, int x) {
    bool valid = false;

    while (n > 9) {
      if (n % 10 == x)
        valid = true;

      n /= 10;
    }

    if (n == x)
      return false;

    return valid;
  }
};

File: easy/3908_valid_digit_number/Solution.java
class Solution {
  public boolean validDigit(int n, int x) {
    int nCopy = n;
    boolean valid = false;

    while (nCopy > 9) {
      if (nCopy == x) {
        valid = true;
      }

      nCopy /= 10;
    }

    return (nCopy != x) && valid;
  }
}
